### PR TITLE
[Widgets] Try inserting intermediate sub-sections in widgets customizer

### LINKED ISSUE #23975: Blocks in Customizer: Current flow is incorrect
**Is your feature request related to a problem? Please describe.**
Similarly to how widgets currently work in the Customizer, the experimental widget-blocks should also work. The widgets require two clicks to get into an individual widget area. When I click "Widgets" I get another menu showing the available widget areas. I click on one of those and then can start adding in widgets to that one area.

When I click on the experimental widget-blocks, it is one click to see both widget-block areas as accordions. I really think there should be another click to get into the individual widget-block area. 

**Screencast**

widgets

**Expectations**
I expect, when clicking into "Widget-blocks (experimental)" to see another slideout menu that displays each widget area.

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Trying to fix #23975.

This PR is an attempt to add intermediate sub-sections to Widgets customizer. This PR still suffers from several issues including #25157, #25154, and #25201.

Still unable to get it working as expected though. The goal is to add the `widget-area` block to the editor, currently I'm adding the inner blocks of the `widget-area` block. However, I can't seem to find a way to add only one `widget-area` block to the editor? What am I missing and how should I fix this?

## How has this been tested?

1. Open the customizer
2. Click "Widgets"
3. View the widgets editor in the customizer in sub-sections.

## Screenshots 
image

image

## Types of changes

New feature

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-widgets/src/index.js [hunk 1]
```diff
@@ -44,15 +46,27 @@ export function initialize( id, settings ) {
...
-	registerCoreBlocks();
-	if ( process.env.GUTENBERG_PHASE === 2 ) {
-		__experimentalRegisterExperimentalCoreBlocks( settings );
+export function customizerInitialize( id, settings, widgetID ) {
+	// The customizer can has many widgets, it should only register blocks once.
+	if ( ! registered ) {
+		registerCoreBlocks();
+		if ( process.env.GUTENBERG_PHASE === 2 ) {
+			__experimentalRegisterExperimentalCoreBlocks( settings );
+		}
```
Comment (adamziel) [inline packages/edit-widgets/src/index.js:57]: We'll also need `registerBlock( legacyWidget );` as in line 33 in the same file. Maybe it's time to that logic into our own `registerWidgetsEditorBlocks()` function?

**Comment by adamziel** [comment]
I left a minor note but otherwise looks good - keep up the good work!

**Comment by adamziel** [comment]
Blocked by #25625

**Comment by noisysocks** [comment]
No longer needed, see https://github.com/WordPress/gutenberg/issues/23975#issuecomment-754384760.

**Pull request closed without merging**